### ISSUE TITLE
Broken feature unification when multiple packages depend on `askama`

### ISSUE THREAD

**Issue by dpc** [djc/askama#810]
I'm trying to make a workspace that in two different places of dependency tree uses `askama`

One imports only `askama`, the other one `askama_axum`. The one that only imports askama fails with:

```
error[E0433]: failed to resolve: could not find `askama_axum` in the list of imported crates                                                                    
   --> /home/dpc/.cargo/git/checkouts/uniffi-rs-5e0d6f877b5a61c0/b24850f/uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs:120:10
    |                                                                                                                                                           
120 | #[derive(Template)]                                                                                                                                       
    |          ^^^^^^^^ could not find `askama_axum` in the list of imported crates
    |                                                                                                                                                           
    = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)                                          
```

Seems to me that the other crate enabled `with-axum` feature, and `askama_derive` is being picked up by the `uniffi-rs` so it now generates code that wants to access `askama_axum`, which is not a dependency of it.

I've read through https://github.com/djc/askama/issues/588#issuecomment-1485986574 which seems somewhat related.

Unless I'm missing something somewhere, this might be a hard circle to square. Downstream crates with different extenson-sets, can't share the `askama_derive`, it seems.

**Comment by djc** [comment]
Yeah, not sure if there's a solution for this. The easiest solution is probably to have both of your crates enable with `with-axum` feature.

**Comment by djc** [comment]
Having to decorate each type with all the integrations needed seems like a pain in the ass. But, we could set this up in the configuration? So you declare in the `askama.toml` for your crate which integrations you want.

**Comment by dpc** [comment]
If that's possible, then it does seem like an improvement.

**Comment by 5iddy** [comment]
I am experiencing this issue as well but with `with-actix-web` feature.
Here is the errror:
```
error[E0433]: failed to resolve: could not find `askama_actix` in the list of imported crates
 --> src/templates/index.rs:3:17
  |
3 | #[derive(Debug, Template)]
  |                 ^^^^^^^^ could not find `askama_actix` in the list of imported crates
  |
  = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0433`.
error: could not compile `xprs` due to previous error
```

**Comment by djc** [comment]
@5iddy you should be pulling the `askama_actix` crate into your dependency graph.

**Comment by qsantos** [comment]
I have encountered the same issue as @dpc. To give some explanation as to why it is non-trivial to solve this:

When the `with-axum` feature is set, the `Template` derive macro will generate some code that depends on `askama_axum`:

https://github.com/djc/askama/blob/b8697ba20205c01ac43da4e3054a2d3f3be007b9/askama_derive/src/generator.rs#L171-L183

This code might not be called by the crate which depends on `askama` and not `askama_axum`. However, when resolving the dependencies at the workspace level, having another crate depending on `askama_axum` will enable the `with-axum` feature (features are meant to be additive only precisely to make this possible). So, we still need to generate the code, so we still need to refer to `askama_axum`.

In other situations, it would just mean adding `askama_axum` as an optional dependency to `askama` that is enabled when the `with-axum` feature is set and re-exporting it (so that the code generated by the macro can find it at `::askama::askama_axum`). However, here, `askama_axum` actually depends **on** `askama`. So, this would be a cyclic dependency, which is not allowed.

In short, making this work as-expected would require a deep rework of the Askama workspace.

**Comment by djc** [comment]
I'm open to reviewing a rework of the Askama workspace if we think there is a way to solve this, but the right way to solve this is not obvious to me. In particular, having all the integrations be Cargo features in the askama crate is not a workable situation -- that is what we had before, but it leads to issues because the Askama crate ends up conditionally depending on a whole lot of other crates some of which might conflict (like, Rocket might depend on a different version of hyper than Axum, which is a pain in the ass to maintain).

The problem here is that we have Askama usage in different crates, and whether or not to generate code for an integration is propagated via Cargo features. Askama crates get compiled with a unified set of features, while the support code necessary for the integration might not available in each crate that has templates in it. However, it's not obvious how procedural macro invocations across crates might communicate with each other (alternatively, how we could block procedural macro invocations across crates from communicating with each other, I suppose?). `askama.toml` as a configuration method functions on a per-crate basis.
